plugins/modules/content_view_version.py

```diff
@@ -230,6 +227,12 @@ def main():
             module.ensure_entity('content_view_versions', None, content_view_version, params=scope)
         else:
             module.auto_lookup_entities()
+
+            # only need to record if not desired_absent, as ensure_entity already does all the proper recording for us
+            content_view_version_record = _flatten_entity(content_view_version, module.foreman_spec)
+            content_view_version_record['environments'] = [environment['name'] for environment in content_view_version_record.get('environments', [])]
```

Comment:
a tad unsure about this `environments` flattening here. it slightly changes the behaviour from what we had before, but the data here is supposed to be "flat" anyway.

Reply:
LGTM :shrug: 

I don't see why Ansible users would need any other lifecycle environment metadata here, other than the names.